### DISCUSSION TITLE
Has ImageRun Been Depreciated?

### DISCUSSION THREAD

**Discussion by MabryLC** [dolanmiu/docx#1320] [Q&A] [answered]
I have been using paragraphs to put an image made from Media.addImage in but, depending on the resolution, (or possibly the app I'm using to take the pictures) I am getting images rotated 90 degrees in my document.

So I decided to look at the documentation and realized I wasn't even doing it right the way it was explained with the ImageRun class; Once I tried it and tried to import the class into my typescript file it gave me the following error: error TS2305: Module '"../../node_modules/docx/build"' has no exported member 'ImageRun'.

Is it possible that ImageRun has been depreciated and just not updated? I noticed there was a class called PictureRun but it requires a byte stream and I have no idea on how to implement that. If it has been depreciated, what can I use? I would really like to be able to make my images display more reliably.

**Comment by dolanmiu** [comment, marked as answer]
No,

ImageRun is not deprecated

Usage can be found here:

https://github.com/dolanmiu/docx/blob/master/demo/5-images.ts

**Reply by dolanmiu** [reply]
Media.addImage is deprecated though

It was complex to use, now its much more simpler

**Reply by MabryLC** [reply]
Thanks for answering me, sir.
